packages/flutter/lib/src/gestures/mouse_tracking.dart

```diff
@@ -26,14 +26,44 @@ typedef PointerExitEventListener = void Function(PointerExitEvent event);
 /// Used by [MouseTrackerAnnotation], [MouseRegion] and [RenderMouseRegion].
 typedef PointerHoverEventListener = void Function(PointerHoverEvent event);
 
+// A key, so that (_IdenticalAnnotationKey(a) == _IdenticalAnnotationKey(b)) is equivalent to
+// identical(a, b).
+class _IdenticalAnnotationKey implements Key {
+  _IdenticalAnnotationKey(this.value);
+
+  final MouseTrackerAnnotation value;
+
+  // Don't return `identityHashCode(value)` directly since the hash of this key
+  // needs to be different from that of `value`. Do a fast conversion with a
+  // random mask, since we don't need a secure hashing algorithm for a private
+  // class.
+  @override
+  int get hashCode => _hashCodeMask ^ identityHashCode(value);
+  static const int _hashCodeMask = 0xc366d8df10ee8;
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other))
+      return true;
+    return other is _IdenticalAnnotationKey && identical(other.value, value);
+  }
+}
+
 /// The annotation object used to annotate layers that are interested in mouse
 /// movements.
 ///
 /// This is added to a layer and managed by the [MouseRegion] widget.
-class MouseTrackerAnnotation {
+class MouseTrackerAnnotation extends Diagnosticable {
   /// Creates an annotation that can be used to find layers interested in mouse
   /// movements.
-  const MouseTrackerAnnotation({this.onEnter, this.onHover, this.onExit});
+  ///
+  /// All arguments are optional.
+  const MouseTrackerAnnotation({
+    this.onEnter,
+    this.onHover,
+    this.onExit,
+    Key key,
+  }) : _key = key;
```

Comment:
why not just assert `key != null`?

Reply:
(then you wouldn't need your magical key type)

Reply:
I think it'll be easier in some cases to have the key always non-null, since `MouseTracker` put `MouseTrackerAnnotation`s in a set by its key. And now it's using the `ObjectKey`.

Reply:
I don't understand the use case here. Other than in tests, you never actually pass null into the `key` argument here. What's problem does it solve to have null be handled by creating an ObjectKey?